components/Boards/ArticleSection.tsx

```diff
@@ -0,0 +1,41 @@
+import { useRouter } from "next/router";
+import { useState, ChangeEvent, FormEvent } from "react";
+import SearchIcon from "@/assets/images/icons/ic_search.svg";
+import styles from "@/components/Boards/ArticleSection.module.css";
+
+export default function ArticleSection() {
+  const router = useRouter();
+  const [value, setValue] = useState<string>("");
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
```

Comment:
아래처럼 이벤트 핸들러 함수들은 `useCallback` 사용해서 메모이제이션하는 방법도 있어요. 이런 경우, 불필요한 리랜더링을 방지해줄 수 있습니다.

```suggestion
  const handleChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  }, []);
```